### DISCUSSION TITLE
prismaSchemaFolder don't work properly

### DISCUSSION THREAD

**Discussion by poyrazhancilar** [prisma/prisma#25606] [Q&A]
### Question

I am trying to use multiple Prisma files with Prisma, and I have reviewed everything, including Prisma's own documentation. However, whenever I run npx prisma generate, I get the following error:

```bash
> npx prisma generate
Environment variables loaded from .env
Prisma schema loaded from src/prisma/schema/schema.prisma
Error: 
You don't have any models defined in your schema.prisma, so nothing will be generated.
You can define a model like this:

model User {
  id    String  @id @default(auto()) @map("_id") @db.ObjectId
  email String  @unique
  name  String?
}

More information in our documentation:
https://pris.ly/d/prisma-schema
```

File & Folder 
<img width="294" alt="Screenshot 2024-11-08 at 15 51 27" src="https://github.com/user-attachments/assets/8ac9740b-43e0-4db8-ac36-0022a53dec5f">

Structure:

schema.prisma:
```prisma
datasource db_main {
  provider = "mongodb"
  url      = env("MONGODB_URI")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["prismaSchemaFolder"]
}
```

contact.prisma
```prisma
datasource db_contact {
  provider = "mongodb"
  url      = env("MONGODB_URI")
}

model ContactRequest {
  id          String  @id @default(cuid()) @map("_id")
  email       String
  message String?

  @@map("contact_requests")
}
```

Why am I getting this error? I have checked the things in the documentation and other sources, but I am still experiencing this issue. I would appreciate it if you could help, thank you.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nurul3101** [comment]
Hello @poyrazhancilar 👋 

Can you try removing datasource block from your `contact.prisma` file? It seems that you have defined datasource already in your `schema.prisma` file. Do you get the same error after removing datasource block?

Your contact.prisma file would look like this:
```prisma
model ContactRequest {
  id          String  @id @default(cuid()) @map("_id")
  email       String
  message String?

  @@map("contact_requests")
}
```

**Reply by poyrazhancilar** [reply]
I just tried it again but unfortunately, I'm still getting the same error.

**Reply by nurul3101** [reply]
What happens if you do `npx prisma generate --schema prisma/schema`?

**Reply by poyrazhancilar** [reply]
I tried it.

```sh
> npx prisma generate --schema src/prisma/schema/schema.prisma
Environment variables loaded from .env
Prisma schema loaded from src/prisma/schema/schema.prisma
Error: 
You don't have any models defined in your schema.prisma, so nothing will be generated.
You can define a model like this:

model User {
  id    String  @id @default(auto()) @map("_id") @db.ObjectId
  email String  @unique
  name  String?
}

More information in our documentation:
https://pris.ly/d/prisma-schema
```

**Reply by ludralph** [reply]
Hi @poyrazhancilar 

Based on the error message you're seeing, it appears that your Prisma schema file (src/prisma/schema/schema.prisma) doesn't contain any model definitions. You need to define a model in your schema.prisma file.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;

**Comment by RangerCoder99** [comment]
This is likely happening because OP's schema.prisma is still inside the main folder It all needs to be moved under ./prisma/schema

**Comment by d30jeff** [comment]
Had the same issue with **Prisma 6.5.0**

Changing from 

```
"prisma": {
  "schema": "prisma/schema/schema.prisma"
},
```

to
```
"prisma": {
  "schema": "prisma/schema"
},
```

in my `package.json` file fixed the problem

**Reply by shoaibhajj09** [reply]
thanks bro ❤
